automation/Remote.psm1

```diff
@@ -36,28 +36,49 @@ function Start-VirtualMachine {
 
         [String]
         [ValidateNotNullOrEmpty()]
-        $ISOFile
+        $ISOFile,
+
+        [String]
+        [ValidateNotNullOrEmpty()]
+        $Platform = "Hyper-V"
     )
 
-    # set the vm switch first
-    $Switch = Set-VmSwitch -SwitchName $SwitchName
+    if($Platform -eq "Hyper-V"){
+        # set the vm switch first
+        $Switch = Set-VmSwitch -SwitchName $SwitchName
+        $VM = @{
```

Comment:
Split HyperV and Virtual box to seperate functions to make it easier to mange each of them

Reply:
ya will do eventually, wanted to push code a bit earlier, will clean up eventually when finished with everything else on this PR